### DISCUSSION TITLE
Handling of NGP ROM name

### DISCUSSION THREAD

**Discussion by ickaze** [sanni/cartreader#708] [Coding Corner]
When I tried to read the Japanese version of the NGP cotton, the ROM name was 'COTTON-NGP/C' and it displayed 'Can't create file' and could not be read because there was a '/' in the file name that could not be used.
Therefore, I recommend the following response to replace '/' with '_' when reading the ROM name.

  // get app name
  for (uint32_t i = 0; i < 17; i++)
  {
    romName[i] = readByte_NGP(0x24 + i);
    if( romName[i] == '/' )
      romName[i] = '_';
  }

**Comment by PsyK0p4T** [comment]
In fact, this sounds logical here. Cotton is maybe the first cart to have this character in header...
I will confirm the fix tonight, and add it to the code then. Thanks for the notice :)

**Comment by PsyK0p4T** [comment]
Issue confirmed, and fix pushed ;) Thank you !

**Comment by PsyK0p4T** [comment]
@ickaze is your Cotton cartridge an original one ?
If it is, I would be interested by your dump result, and the cart info logs too. This game is one of the rarest to find on NGP, and its hashes still need to be confirmed on datomatic. Can you message me on the OSCR Discord please ?

**Reply by ickaze** [reply]
Yes, it's original. I am not certain when I purchased it, but based on when it was released, I think I pre-ordered it and bought it on the day it was released.

The hash value and header of the 1Mbytes rom cartridge were as follows, but if you have any other information you need, please invite ickaze#0707 to Discord.

MD5:
	a123c495ec561f4c061e96ea29a41299
SHA-1:
	371a607d93ee4ea0d3824458bf163f80c5a6bd16
SHA-256:
	c1fd1d075b2cc088c62898c7cf00c2686e701f0ed321710bed2ae970be41d2f0

--------|-0--1--2--3--4--5--6--7--8--9--A--B--C--D--E--F-|---0123456789ABCDEF
00000000|20 4c 49 43 45 4e 53 45 44 20 42 59 20 53 4e 4b |    LICENSED BY SNK
00000010|20 43 4f 52 50 4f 52 41 54 49 4f 4e 00 80 2d 00 |    CORPORATION..-.
00000020|80 00 10 10 43 4f 54 54 4f 4e 2d 4e 47 50 2f 43 |   ....COTTON-NGP/C
00000030|00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |   ................

**Reply by PsyK0p4T** [reply]
hmm yes please, I would need more things. I'm adding you now :)
